src/probe.rs

```diff
@@ -149,41 +149,62 @@ impl<R: Read + Seek> Probe<R> {
 
 	#[allow(clippy::shadow_unrelated)]
 	fn guess_inner(&mut self) -> Result<Option<FileType>> {
+		// temporary buffer for storing 36 bytes
+		// (36 is just a guess as to how long the data for estimating the file type might be)
 		let mut buf = [0; 36];
 
-		let pos = self.inner.seek(SeekFrom::Current(0))?;
+		// read the first 36 bytes and seek back to the starting position
+		let starting_position = self.inner.stream_position()?;
 		let buf_len = std::io::copy(
-			&mut self.inner.by_ref().take(36),
+			&mut self.inner.by_ref().take(buf.len() as u64),
 			&mut Cursor::new(&mut buf[..]),
 		)? as usize;
+		self.inner.seek(SeekFrom::Start(starting_position))?;
 
-		self.inner.seek(SeekFrom::Start(pos))?;
-
+		// estimate the file type by using these 36 bytes
+		// note that any error from `from_buffer_inner` are suppressed, as it returns an error on unknown format
+		// - TODO: why is the case `Err(LoftyError::UnknownFormat)` suppressed, but only for `from_buffer_inner`?
+		// - What is the special meaning of the return type `Ok(None)` vs `Err(LoftyError::UnknownFormat)`?
 		match FileType::from_buffer_inner(&buf[..buf_len]) {
+			// the file type was guessed based on these bytes
 			Ok((Some(f_ty), _)) => Ok(Some(f_ty)),
+			// the first data block is ID3 data; this means other data can follow (e.g. APE or MP3 frames)
 			Ok((None, id3_len)) => {
-				self.inner
+				// the position right after the ID3 block is the internal size value (id3_len)
+				// added to the length of the ID3 header (which is 10 bytes),
+				// as the size does not include the header itself
+				let position_after_id3_block = self
+					.inner
 					.seek(SeekFrom::Current(i64::from(10 + id3_len)))?;
 
-				let mut ident = [0; 3];
-				let buf_len = std::io::copy(
-					&mut self.inner.by_ref().take(3),
-					&mut Cursor::new(&mut ident[..]),
-				)?;
-
-				self.inner.seek(SeekFrom::Start(pos))?;
-
-				if buf_len < 3 {
-					return Err(LoftyError::UnknownFormat);
-				}
-
-				if &ident == b"MAC" {
-					Ok(Some(FileType::APE))
-				} else if verify_frame_sync([ident[0], ident[1]]) {
-					Ok(Some(FileType::MP3))
-				} else {
-					Err(LoftyError::UnknownFormat)
-				}
+				let file_type_after_id3_block = {
+					// try to guess the file type after the ID3 block by inspecting the first 3 bytes
+					let mut ident = [0; 3];
+					let buf_len = std::io::copy(
+						&mut self.inner.by_ref().take(ident.len() as u64),
+						&mut Cursor::new(&mut ident[..]),
+					)?;
+
+					if buf_len < 3 {
+						Err(LoftyError::UnknownFormat)
+					} else if &ident == b"MAC" {
```

Comment:
```suggestion
					if &ident == b"MAC" {
```

We don't need this check anymore